### PR TITLE
Replace "klassy" with ES6 classes

### PR DESCRIPTION
**Pull request by Turbo87**
/cc @trentmwillis @rwjblue @dgeb @stefanpenner

### CODE REVIEW

**Comment by rwjblue** [comment]
Will take some time to do a more thorough review, but I definitely approve the effort (and from prior conversations with @dgeb I believe he does as well).

**Comment by Turbo87** [comment]
💚  ... thanks Travis :)

**Comment by dgeb** [comment]
Yeah, Klassy is a lib I will gladly retire. Thanks @Turbo87.

File: lib/ember-test-helpers/test-module-for-acceptance.js [hunk 1]
```diff
@@ -2,18 +2,18 @@ import AbstractTestModule from './abstract-test-module';
 import Ember from 'ember';
 import { getContext } from './test-context';
 
-export default AbstractTestModule.extend({
+export default class extends AbstractTestModule {
   setupContext() {
-    this._super({ application: this.createApplication() });
-  },
+    super({ application: this.createApplication() });
```
Comment (rwjblue) [inline lib/ember-test-helpers/test-module-for-acceptance.js]: This should be `super.setupContext` (bare `super` is only valid in `constructor`).

File: lib/ember-test-helpers/test-module-for-acceptance.js [hunk 2]
```diff
@@ -2,18 +2,18 @@ import AbstractTestModule from './abstract-test-module';
...
+  }
 
   teardownContext() {
     Ember.run(() => {
       getContext().application.destroy();
     });
 
-    this._super();
-  },
+    super();
```
Comment (rwjblue) [inline lib/ember-test-helpers/test-module-for-acceptance.js]: Should be `super.teardownContext`

File: lib/ember-test-helpers/test-module-for-component.js [hunk 3]
```diff
@@ -133,18 +136,18 @@ export default TestModule.extend({
...
-      return setupComponentIntegrationTest;
+      return setupComponentIntegrationTest.apply(this, arguments);
     }
-  })(),
+  }
 
-  setupContext: function() {
-    this._super.call(this);
+  setupContext() {
+    super();
```
Comment (rwjblue) [inline lib/ember-test-helpers/test-module-for-component.js]: bare `super` is only valid in `constructor`

File: lib/ember-test-helpers/test-module-for-integration.js [hunk 4]
```diff
@@ -93,7 +95,7 @@ export default AbstractTestModule.extend({
       return container.lookupFactory(subjectName);
     };
 
-    this._super({
+    super({
```
Comment (rwjblue) [inline lib/ember-test-helpers/test-module-for-integration.js]: bare `super` is only valid in `constructor`

File: lib/ember-test-helpers/test-module.js [hunk 5]
```diff
@@ -94,29 +90,29 @@ export default AbstractTestModule.extend({
...
+  setupContext() {
     var subjectName = this.subjectName;
     var container = this.container;
 
     var factory = function() {
       return container.lookupFactory(subjectName);
     };
 
-    this._super({
+    super({
```
Comment (rwjblue) [inline lib/ember-test-helpers/test-module.js]: bare `super` is only valid in `constructor`

Comment (Turbo87) [inline lib/ember-test-helpers/test-module.js]: ouch, forgot about that... thanks!

**Comment by Turbo87** [comment]
@rwjblue done!

**Comment by Turbo87** [comment]
@rwjblue I do agree that those need updates to remove `klassy`, but I don't see why they would error? The way I see it we would just include an unnecessary dependency if we didn't update them, but everything should still be running the same way as before.

**Comment by Turbo87** [comment]
also I tend to say that `klassy` is an implementation detail and as such private API so no need for a major version bump, but we obviously will have to test whether the new `ember-test-helpers` version would run smoothly with older testing libs like `ember-cli-qunit`.

**Comment by rwjblue** [comment]
> I don't see why they would error?

When this PR lands the `klassy` dep will no longer be present, those calls to `resolve.sync('klassy', { baseDir: emberTestHelpersPath })` will begin to error out.

**Comment by rwjblue** [comment]
> I tend to say that klassy is an implementation detail and as such private API

I agree in general. 

> no need for a major version bump

Disagree, for the reason mentioned above. I believe that existing versions of ember-qunit / ember-cli-qunit will error after a `rm -rf node_modules && npm install`.

**Comment by Turbo87** [comment]
> I believe that existing versions of ember-qunit / ember-cli-qunit will error after a rm -rf node_modules && npm install.

I just tried it on a fresh app with `npm link ember-test-helpers` and both testing libs and didn't see any errors

**Comment by rwjblue** [comment]
I don't this npm link is good enough. You would need to ensure there is no version of `klassy` present in `node_modules` (which would not be the case if you installed normally then `npm link`ed).

**Comment by Turbo87** [comment]
> You would need to ensure there is no version of klassy present in node_modules (which would not be the case if you installed normally then npm linked).

you're right, I'm stupid... I see how this would break now. Since `klassy` was a dependency of `ember-test-helpers` and not of `ember-cli-qunit` the resolver call would fail. Since we're sub-1.0 this will only require a minor version bump though to comply with semver.

**Comment by rwjblue** [comment]
> Since we're sub-1.0 this will only require a minor version bump though to comply with semver.

Ya. I was thinking 0.6.0 (I sometimes incorrectly refer to this as a "major" bump when sub 1.0, but what I'm really meaning is that we need to *not* match for users with `^0.5.x`).

**Comment by Turbo87** [comment]
🎉 

will adjust the other libs to account for this change later today unless someone else beats me to it 😉

**Pull request merged**